server/src/main/java/org/elasticsearch/plugins/PluginsUtils.java

```diff
@@ -77,43 +79,113 @@ public static List<Path> findPluginDirs(final Path rootPath) throws IOException
      * Verify the given plugin is compatible with the current Elasticsearch installation.
      */
     public static void verifyCompatibility(PluginDescriptor info) {
-        if (info.isStable()) {
-            if (info.getElasticsearchVersion().major != Version.CURRENT.major) {
+        boolean buildSupportsStablePlugins;
+        SemanticVersion currentElasticsearchVersion = null;
+        try {
+            currentElasticsearchVersion = SemanticVersion.create(Build.current().version());
+            buildSupportsStablePlugins = true;
+        } catch (IllegalArgumentException e) {
+            buildSupportsStablePlugins = false;
+        }
+        // stable plugins can run on the exact version they're built with even if it's not semantic
```

Comment:
Following the discussion in https://github.com/elastic/elasticsearch/pull/100713#discussion_r1358927861, we can assume that the plugin descriptor returns a semantic ES version, so we can simplify this a bit more.